### PR TITLE
Pigeonhole sort in python and java

### PR DESCRIPTION
**Pull request by p-rit**
#746 solved

### CODE REVIEW

File: Pigeonhole_Sort/pigeonhole.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
+public class PigeonholeSort{
```
Comment (jainaman224) [inline Pigeonhole_Sort/pigeonhole.java]: FIx file name.

File: Pigeonhole_Sort/pigeonhole.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# input should be list of integers
```
Comment (jainaman224) [inline Pigeonhole_Sort/pigeonhole.py]: Fix file name.

File: Pigeonhole_Sort/pigeonhole.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        System.out.print(test[i] + " ");
+     }
+}
+
+//input = 1,4,32,2
+// output = 1 2 4 32
+
+            
+        
+
```
Comment (jainaman224) [inline Pigeonhole_Sort/pigeonhole.java]: Please leave only one new line at the end of file.

File: Pigeonhole_Sort/README.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+Pigeonhole sorting is a sorting algorithm that is suitable for sorting lists of elements where the number of elements (n) and the length of the range of possible key values (N) are approximately the same. It requires O(n + N) time. It is similar to counting sort, but differs in that it "moves items twice: once to the bucket array and again to the final destination whereas counting sort builds an auxiliary array then uses the array to compute each item's final destination and move the item there.
+
+### The pigeonhole algorithm works as follows:
+
+* Given an array of values to be sorted, set up an auxiliary array of initially empty "pigeonholes," one pigeonhole for each key through the range of the original array.
+* Going over the original array, put each value into the pigeonhole corresponding to its key, such that each pigeonhole eventually contains a list of all values with that key.
+* Iterate over the pigeonhole array in order, and put elements from non-empty pigeonholes back into the original array.
+
+ ![]( https://github.com/p-rit/Images/blob/master/Pigeonhole_sort_images/Capture.PNG )
+## Pseudocode
```
Comment (jainaman224) [inline Pigeonhole_Sort/README.md]: Add a new line above this.

**Comment by jainaman224** [comment]
@p-rit Are you confused or facing any issue?

**Comment by p-rit** [comment]
I have updated the files as per review @jainaman224

**Comment by jainaman224** [comment]
There are indentation issues. Please use 4 space indentation.

**Comment by p-rit** [comment]
@jainaman224 updated

File: Pigeonhole_Sort/PigeonholeSort.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
+public class PigeonholeSort{
+    // Method for getting the maximum value
+    public static int getMax(int[] inputArray){ 
+                int maxValue = inputArray[0];
```
Comment (jainaman224) [inline Pigeonhole_Sort/PigeonholeSort.java]: Fix indentation throughout the file. There is a shortcut for most editors for auto indent.

File: Pigeonhole_Sort/README.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+* Given an array of values to be sorted, set up an auxiliary array of initially empty "pigeonholes," one pigeonhole for each key through the range of the original array.
+* Going over the original array, put each value into the pigeonhole corresponding to its key, such that each pigeonhole eventually contains a list of all values with that key.
+* Iterate over the pigeonhole array in order, and put elements from non-empty pigeonholes back into the original array.
+
+ ![]( https://github.com/p-rit/Images/blob/master/Pigeonhole_sort_images/Capture.PNG )
+
+ ## Pseudocode
+
+```` find max and min from the array list
+     holeRange := max – min +1
```
Comment (jainaman224) [inline Pigeonhole_Sort/README.md]: Fix indentation.

**Comment by p-rit** [comment]
@jainaman224 solved indentation issues.

**Pull request merged**